### PR TITLE
[strings] Added Euskara (eu) translations

### PR DESCRIPTION
**Pull request by dvdmrtnz**
I added Euskara (eu) translations on `data/categories.txt`, `data/strings/strings.txt`, and `data/strings/types_strings.txt`.

I tried to run the `./tools/unix/generate_localizations.sh` script but no files were generated with the Euskara translations. I guess it is because it is a new language. 

What else do I need to do to add Euskara as a new language?

### CODE REVIEW

**Comment by biodranik** [comment]
Many other files should be changed as well.

Is basque language supported in iOS and Android? How to enable it?

**Comment by pastk** [comment]
> 
> What else do I need to do to add Euskara as a new language?

For iOS please check this commit https://github.com/organicmaps/organicmaps/pull/912/files#diff-971cf1e5ff9bebd47aff357857614bea60d35aeb08d592989b47582266e98d80
Probably you need to do it via Xcode.. @biodranik can advice the details

For android you need to create a localization folder for Euskara manually first (`android/res/values-eu` I guess).
Or run Twine with `--create-folders` options (its run in the bottom of the `tools/unix/generate_localizations.sh` script).

**Comment by dvdmrtnz** [comment]
> Many other files should be changed as well.
> 
> Is basque language supported in iOS and Android? How to enable it?

Yes, it works both on iOS and Android.

For example, here is the Android system language selection:



And here is the iOS app language selection for OsmAnd:

**Comment by biodranik** [comment]
There are several places where new translations should be enabled. But to do it, please add all missing translations into files listed here: https://github.com/organicmaps/organicmaps/blob/master/docs/TRANSLATIONS.md

**Comment by dvdmrtnz** [comment]
I added translations on:

* `data/categories_cuisines.txt`
* `data/strings/sounds.txt`
* `iphone/plist.txt`
* `data/countries_names.txt`

Any other file that would be translated?

File: data/categories_cuisines.txt [hunk 1]
```diff
@@ -1414,6 +1457,7 @@ sk:Vegánske jedlá|Vegetariánske jedlá
 # pt-BR:Frango
 # ro:Pui
 # es:Pollos
+# eu:oiloak
```
Comment (pastk) [inline data/categories_cuisines.txt:1460]: Should start from a capital?

**Review by pastk — COMMENTED** [review]
I checked the files briefly.
Wow, that's a tremendous work, thanks!!

File: data/categories_cuisines.txt (hunk 1 above)
Comment (dvdmrtnz) [reply data/categories_cuisines.txt]: Fixed! Thanks!

**Comment by pastk** [comment]
A new language should be added into `indexer/categories_holder.hpp` definitely and also folders holding android and ios localization files should be created manually (see my previous posts here).

**Comment by dvdmrtnz** [comment]
I added eu to `indexer/categories_holder.hpp` and added the iOS and Android localization folders. 

However, I don’t have access to a macOS machine so I can’t add the language on the Xcode project.

File: indexer/categories_holder.hpp [hunk 2]
```diff
@@ -113,6 +113,7 @@ class CategoriesHolder
       {"vi", 36},
       {"zh-Hans", kSimplifiedChineseCode},
       {"zh-Hant", kTraditionalChineseCode},
+      {"eu", 39},
```
Comment (biodranik) [inline indexer/categories_holder.hpp:116]: Let's sort them by name. I know, it's annoying, we'll refactor it in the future.

**Review by biodranik — COMMENTED** [review]
Thanks a lot for your work! I'll enable the language in iOS before the merge.

You made searchable categories (like food, gas, etc.) in categories.txt and strings.txt equal, right?

Can you please also try to download, unzip and install this debug apk to test if your changes are working correctly? https://github.com/organicmaps/organicmaps/suites/6064429391/artifacts/209481843

**Comment by dvdmrtnz** [comment]
> Thanks a lot for your work! I'll enable the language in iOS before the merge.
> 
> You made searchable categories (like food, gas, etc.) in categories.txt and strings.txt equal, right?

Yes I did make sure that they are the same 


> Can you please also try to download, unzip and install this debug apk to test if your changes are working correctly? https://github.com/organicmaps/organicmaps/suites/6064429391/artifacts/209481843

Translations seem to be working fine except for the countries/regions names on the download maps section of the app. They are showing in English. Why could this be?

**Comment by biodranik** [comment]
> Translations seem to be working fine except for the countries/regions names on the download maps section of the app. They are showing in English. Why could this be?

Please create a corresponding eu folder in data/countries-strings/ and run tools/unix/generate_localizations.sh, these folders are not created automatically to avoid generation of partially translated and not fully enabled languages.

**Comment by biodranik** [comment]
Also create `eu.json` folder in `data/sound-strings/` and regenerate translations.

File: indexer/categories_holder.hpp (hunk 2 above)
Comment (dvdmrtnz) [reply indexer/categories_holder.hpp]: Done

**Review by pastk — APPROVED** [review]

**Comment by biodranik** [comment]
I've created PR with iOS fixes https://github.com/dvdmrtnz/organicmaps/pull/1

**Review by pastk — APPROVED** [review]

**Comment by biodranik** [comment]
Thank you very much for your contribution! We already started maps generation, so eu lang on the map will not appear in the upcoming update, only in the next one.

**Comment by dvdmrtnz** [comment]
Thanks! You guys are making an amazing app! I’m happy to help! 😊

**Pull request merged**